### ISSUE TITLE
Synthesized annotation toString() doesn't match non-synthesized annotation on Java 9+

### ISSUE THREAD

**Issue by jeff-miller-cfa** [spring-projects/spring-framework#28015]
Synthesized annotations that contain arrays and/or strings do not generate the same `toString()` value as the non-synthesized version. 

The non-synthesized annotation's `toString()` method returns a string which is a code equivalent version of that annotation meaning that you can take the result of the `toString()`, copy and paste it into code, and have it compile. This is very convenient when writing a code generation library since it allows you to copy annotations.

This does not work with synthesized annotations though as the way the array gets serialized in `toString()` doesn't match.

Actual Annotation:

```java
@Bean("testValue")
```

Non-Synthesized Annotation `toString()`:

```java
@org.springframework.context.annotation.Bean(value = {"testValue"})
```

Synthesized Annotation `toString()`:

```java
@org.springframework.context.annotation.Bean(value = [testValue])
```

https://github.com/spring-projects/spring-framework/blob/a0c97e4c36e5e07bc13bab4409ec740332a57871/spring-core/src/main/java/org/springframework/core/annotation/SynthesizedMergedAnnotationInvocationHandler.java#L202

**Comment by sbrannen** [comment]
This is actually by design.

The result of invoking `toString()` on a synthesized annotation is identical to the result of invoking `toString()` on the same non-synthesized annotation on Java 8. In fact we used to have tests in place to verify that, but those broke once we started building the framework on Java 9, since Java 9 changed the output format for `toString()` in annotations.

See:

- #19211
- commit faf6e5d8fa05d25504a2f8dcd1ced4931aa82f11

In any case, it does not appear that the JDK format has changed since Java 9, so I'll go ahead and update to the newer format.
